schema_salad/avro/schema.py

```diff
@@ -122,26 +122,15 @@ def __init__(self, name_attr, space_attr, default_space):
         @ard default_space: the current default space or None.
         """
         # Ensure valid ctor args
-        if not (isinstance(name_attr, str) or (name_attr is None)):
-            fail_msg = "Name must be non-empty string or None."
-            raise SchemaParseException(fail_msg)
-        elif name_attr == "":
-            fail_msg = "Name must be non-empty string or None."
-            raise SchemaParseException(fail_msg)
-
-        if not (isinstance(space_attr, str) or (space_attr is None)):
-            fail_msg = "Space must be non-empty string or None."
-            raise SchemaParseException(fail_msg)
-        elif name_attr == "":
-            fail_msg = "Space must be non-empty string or None."
-            raise SchemaParseException(fail_msg)
-
-        if not (isinstance(default_space, str) or (default_space is None)):
-            fail_msg = "Default space must be non-empty string or None."
-            raise SchemaParseException(fail_msg)
-        elif name_attr == "":
-            fail_msg = "Default must be non-empty string or None."
+        def validate(val, name):
```

Comment:
This needs type hints

Reply:
Sorry - forgot to run tox! Thanks for the speedy comment. Fixed in push.